qulacs統合のお知らせを追加 (#364)
Qulacs-Osaka/qulacs-osaka: README.md

## Relation to [Qulacs](https://github.com/qulacs/qulacs)
The commits in Qulacs are merged into Qulacs-Osaka up to the pull request corresponding to the commit  `987474b31a6e60eba116d2e40ab538dcf8086038` ([link to the corresponding commit in Qulacs](https://github.com/qulacs/qulacs/commit/987474b31a6e60eba116d2e40ab538dcf8086038) and [in Qulacs-Osaka](https://github.com/qulacs/qulacs/commit/987474b31a6e60eba116d2e40ab538dcf8086038)). After that commit, this project is developed independently from Qulacs and has many new features that have not been available in Qulacs.

> 

Reviewer: 内容は、下で問題無いですが、リポジトリに訪れた人が気がつくように、READMEの上部にもリンクがあれば分かりやすいかなと思いました。
blockquoteの記法が使えそうですが、別にこれは使わなくてもいいです。
https://github.com/github-community/community/discussions/16925

```
> **Note**
> Qulacs-Osaka/qulacs-osaka will be integrated into the qulacs/qulacs. For more details, please refer to [Information](#Information) section.
```

Author: 入れときます！

Author: @Hiroya-W 入りました。

## How to cite

Please cite this arXiv paper: [Qulacs: a fast and versatile quantum circuit simulator for research purpose](https://arxiv.org/abs/2011.13524)

> ## Infomation

Reviewer: typoです

```suggestion
## Information
```

> Experimental new features of Qulacs that have been developed in the Osaka University repository [Qulacs-Osaka/qulacs-osaka](https://github.com/Qulacs-Osaka/qulacs-osaka) will be integrated into the original [Qulacs](https:// github.com/qulacs/qulacs). The following new features will be added!!!

Reviewer: URLに余計な空白がありました。

```suggestion
Experimental new features of Qulacs that have been developed in the Osaka University repository [Qulacs-Osaka/qulacs-osaka](https://github.com/Qulacs-Osaka/qulacs-osaka) will be integrated into the original [Qulacs](https://github.com/qulacs/qulacs). The following new features will be added!!!
```

Author: Thx!

### Integration date
Scheduled around August 2022.

### New features
The main new features are as follows

> - Provision of type hint files for python

Reviewer: Pythonの頭文字は大文字にしておきたいです。
provisionって、サーバーの環境構築を指すイメージを持っていますが、ヒントファイルの提供なら、provideの方が合いそう？と思いました

```suggestion
- Providing type hint files for Python
```

Author: Thank you!

> 	- Prevents runtime errors due to wrong argument types.
> 	- Configure tools such as mypy to take full advantage of type hint information.

Reviewer: Pythonは型ヒントファイルがあったとしても、引数に違う型を与えていても動いてしまうんでしたね...。
こんな感じでどうでしょうか？

```suggestion
	- Configure tools such as mypy to take full advantage of type hint information.
	- mypy can detect the use of incorrect argument types in the qulacs API.
```

- Sending exceptions with detailed information
	- Makes it easier to understand the cause of the error.

> 	- (For jupyter notebook users) kernel is less likely to crash if incorrect input is given.

Reviewer: 同様に頭文字を大文字にします

```suggestion
	- (For Jupyter Notebook users) kernel is less likely to crash if incorrect input is given.
```

- Added backprop (gradient calculation by error back propagation method) to ParametricQuantumCircuit
	- It is faster than calculating gradients one by one.
- Gradient Calculator

### Scope of impact
The existing functionality has not been changed, so the existing code using Qulacs will continue to work as is. However, since the implementation language of csim has been changed from C to C++, changes may be necessary if you have been using direct calls to csim.

> Due to the C++ change, all complex numbers are now handled by std::complex.

Reviewer: コードスパンで囲んでおくと分かりやすいかなと思いました。

```suggestion
Due to the C++ change, all complex numbers are now handled by `std::complex`.
```

### Add dependency libraries

> This integration adds boost as a dependent library.

Reviewer: セクション名で dependency libraries と使っているので、同じように使う方が読みやすいかなと思いました。

```suggestion
This integration adds boost as a dependency library.
```
